File: src/main/java/com/drunkensoftball/api/exception/NonUniqueException.java
package com.drunkensoftball.api.exception;

import org.springframework.web.bind.annotation.ResponseStatus;

import static org.springframework.http.HttpStatus.CONFLICT;

@ResponseStatus(CONFLICT)
public class NonUniqueException extends RuntimeException {

    public NonUniqueException(final String message){
        super(message);
    }

    public NonUniqueException(final String message, final Throwable cause){
        super(message, cause);
    }
}

File: src/main/kotlin/com/drunkensoftball/api/exception/NonUniqueException.kt
package com.drunkensoftball.api.exception

import org.springframework.web.bind.annotation.ResponseStatus

import org.springframework.http.HttpStatus.CONFLICT

@ResponseStatus(CONFLICT)
class NonUniqueException : BaseDSException {

    constructor(message: String) : super(message) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}
}
